pkg/clients/filter.go

```diff
@@ -67,6 +67,9 @@ func (f *Filter) FilterIncidents(incidents []pagerduty.Incident) []pagerduty.Inc
 
 		if keep {
 			res = append(res, inc)
+			if len(res) == int(f.GetLimit()) {
```

Comment:
Why are results being dropped here? The `limit` is passed to the pagerduty API.

Reply:
No, the limit isn't given to the PD API, because this was just an absolute amount. Means: 1 was default (latest) you just get one Incident, which necessarily isn't the one you're looking for. The content filtering is done here - so the limit on how much you want to get back needs to be checked here.

Reply:
Then the limit should be passed to the PagerDuty API if it's set. Dropping results we already have makes no sense here. I suggest to remove from this and handle in a separate PR.

Reply:
Not sure if we do have a misunderstanding.

old flow:  pdClient.ListIncidents(filter just triggered and limit - not alert text) > local filter of result if it match alert text

related code :

- filter limit set - you want to get the latest:
https://github.com/sapcc/pulsar/blob/faed78895e4eb0cb8636f0e95883641f14e84072/pkg/clients/pagerduty.go#L127-L129

- filter used for PD (not alert specific) - and with that you just get any alert:
https://github.com/sapcc/pulsar/blob/faed78895e4eb0cb8636f0e95883641f14e84072/pkg/clients/pagerduty.go#L101-L105
	
- post filter incidents on text to get from the list our specific alert):
https://github.com/sapcc/pulsar/blob/faed78895e4eb0cb8636f0e95883641f14e84072/pkg/api/acknowledge.go#L66-L68

my change gets all open incidents and stops the filter loop if the latest alert is found.
didn't see a text filter in the PD api - guess this is why you did the post filtering.